lib/service/discovery.go

```diff
@@ -122,3 +139,42 @@ func (process *TeleportProcess) initDiscoveryService() error {
 func (process *TeleportProcess) integrationOnlyCredentials() bool {
 	return process.Config.Auth.Enabled && modules.GetModules().Features().Cloud
 }
+
+// buildAccessGraphFromTAGOrFallbackToAuth builds the AccessGraphConfig from the Teleport Agent configuration or falls back to the Auth server's configuration.
+// If the AccessGraph configuration is not enabled locally, it will fall back to the Auth server's configuration.
+func buildAccessGraphFromTAGOrFallbackToAuth(ctx context.Context, config *servicecfg.Config, client auth.ClientI, logger *slog.Logger) (discovery.AccessGraphConfig, error) {
+	var (
+		accessGraphCAData []byte
+		err               error
+	)
+	if config.AccessGraph.CA != "" {
+		accessGraphCAData, err = os.ReadFile(config.AccessGraph.CA)
+		if err != nil {
+			return discovery.AccessGraphConfig{}, trace.Wrap(err, "failed to read access graph CA file")
+		}
+	}
+	accessGraphCfg := discovery.AccessGraphConfig{
+		Enabled:  config.AccessGraph.Enabled,
+		Addr:     config.AccessGraph.Addr,
+		Insecure: config.AccessGraph.Insecure,
+		CA:       accessGraphCAData,
+	}
+	if !accessGraphCfg.Enabled {
+		logger.Debug("Access graph is disabled or not configured. Falling back to the Auth server's access graph configuration.")
+		ctx, cancel := context.WithTimeout(ctx, 5*time.Second)
+		rsp, err := client.GetClusterAccessGraphConfig(ctx)
+		cancel()
+		switch {
+		case trace.IsNotImplemented(err):
+			logger.Debug("Auth server does not support access graph's GetClusterAccessGraphConfig RPC")
+		case err != nil:
+			return discovery.AccessGraphConfig{}, trace.Wrap(err)
+		default:
+			accessGraphCfg.Enabled = rsp.Enabled
```

Comment:
nit: Having a `Clone()` method would make any config changes easier as we copy all fields in multiple places now.

Reply:
this copies to discovery config settings. A we would need to have multiple clones with different values per target